src/main/java/florizz/objects/Bouquet.java

```diff
@@ -31,4 +44,70 @@ public boolean equals(Object obj) {
         // Compare the data members and return accordingly
         return Objects.equals(c.bouquetName.toUpperCase(), this.bouquetName.toUpperCase());
     }
+
+    /**
+     * check if flower exist in bouquet.
+     *
+     * @param flowerName
+     * @return boolean true if flower exist
+     */
+    public boolean doesFlowerExist(Flower flowerName) {
+        if (flowerHashMap.get(flowerName) == null) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * add flowers into bouquet
+     *
+     * @param flowerName
+     * @param quantity
+     */
+    public void addFlower(Flower flowerName, Integer quantity) {
+        if (doesFlowerExist(flowerName)) {
+            Integer currentQuantity = getFlowerQuantity(flowerName);
+            Integer newQuantity = currentQuantity + quantity;
+            flowerHashMap.replace(flowerName, newQuantity);
+        } else {
+            flowerHashMap.put(flowerName,quantity);
+        }
+    }
+
+    public boolean removeFlower(Flower flowerName, Integer quantity) {
+        // if flower already in bouquet
+        if (doesFlowerExist(flowerName)) {
+            Integer currentQuantity = getFlowerQuantity(flowerName);
+            Integer newQuantity = currentQuantity - quantity;
+            if (newQuantity < 0) {
+                System.out.println("Tried to remove more than quantity available, quantity set to 0");
```

Comment:
can consider throwing an exception instead?

Reply:
Will exception works? Since the programme continues to assign `newQuantity = 0` instead of stopping